File: Kapitel_11_bis_16/16_03_C_und_C++/wiealtbinich.c
// Enthaelt Funktionen um auf die Konsole zu schreiben und von ihr zu lesen
#include <stdio.h>

// Funktionen um das Datum zu ermitteln
#include <time.h>


void berechnen()
{		
	// Ermittlung der aktuellen Systemzeit
	time_t rohZeit = time(NULL);
	
	// Umrechnung in ein konkretes Datumsformat
	struct tm *formatiertesDatum = localtime(&rohZeit);
	
	// Ausgabe an den Nutzer
	printf("In welchem Jahr wurden Sie geboren?\n");
		
	// Eingabe des Nutzers in Variable speichern
	int jahresanzahl;	
	scanf("%d", &jahresanzahl);
		
	// Jahre errechnen
	int alter = formatiertesDatum->tm_year - jahresanzahl + 1900;	

	// Validierung der Berechnung
	if (alter < 0 || alter > 120)
	{
		printf ("Bitte geben Sie Ihr Geburtsjahr korrekt ein.");
		// Erneuter Aufruf falls ein Fehler vorliegt
    	berechnen();
	}    	
  	else
	{		
		// Ausgabe des Alters
		printf("Sie sind %d Jahre jung.", alter);
	}    
}

// Aufruf der Berechnung nach Programmstart
int main(int argc, char* argv[]) {
	
	berechnen();
	
	return 0;
}

File: Kapitel_11_bis_16/16_06_Python/wiealtbinich.py
# Laden der Standardfunktionen zu Datumsberechnung
from time import *

# Anlegen einer neuen Funktion
def berechnen():
	# Ermitteln der aktuellen Jahreszahl
	heute = gmtime().tm_year

	# Einlesen der Eingabe
	eingabe = input("In welchem Jahr wurden Sie geboren?\n")

    # Konvertieren der Eingabe in eine Zahl
	jahr = int(eingabe)

    # Berechnung des Datums
	alter = heute - jahr

    # Validierung
	if alter < 0 or alter > 120:
		print("Bitte geben Sie Ihr Geburtsjahr korrekt ein.")
		berechnen()
	else:
		# Ausgabe des Datums
		print("Sie sind",alter,"Jahre jung.")

#Aufruf der Methode sobald das Script gelesen wird
berechnen()